crates/sncast/tests/e2e/declare.rs

```diff
@@ -286,12 +288,75 @@ async fn test_many_packages_default() {
         "99999999999999999",
     ];
 
-    // todo: (1247) Execute tests in temp dirs
     let snapbox = Command::new(cargo_bin!("sncast"))
-        .current_dir(CONTRACTS_DIR.to_string() + "/multiple_packages")
+        .current_dir(tempdir.path())
         .args(args);
     snapbox.assert().failure().stderr_matches(indoc! {r"
         ...
         Error: More than one package found in scarb metadata - specify package using --package flag
     "});
 }
+
+#[tokio::test]
+async fn test_worskpaces_package_specified_virtual_fibonacci() {
+    let tempdir = copy_directory_to_tempdir(CONTRACTS_DIR.to_string() + "/virtual_workspace");
+    let accounts_json_path = get_accounts_path("tests/data/accounts/accounts.json");
+    let args = vec![
+        "--url",
+        URL,
+        "--accounts-file",
+        accounts_json_path.as_str(),
+        "--account",
+        "user8",
+        "--int-format",
+        "--json",
+        "declare",
+        "--package",
+        "cast_fibonacci",
+        "--contract-name",
+        "FibonacciContract",
+        "--max-fee",
```

Comment:
[nitpick]: in tests, we can use auto_estimated `max-fee`. IMO it will be cleaner.

Reply:
most of the time it won't work sadly :( the problem is that we execute most of the tests concurrently, and the order of them is dufferent from run to runu - that is why the fee price fluctuates, big time. This may lead to non deterministic tests, failing from time to time because the estimated fee that was valid a second ago, is now too little.